go/libraries/doltcore/sqle/types/uuid.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/src-d/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+type uuidType struct{}
+
+func (uuidType) Value() types.Value {
+	return types.KindToType[types.UUIDKind]
+}
+
+func (uuidType) PrimaryType() sql.Type {
+	return sql.Text
+}
+
+func (uuidType) RepresentingTypes() []sql.Type {
+	return []sql.Type{}
```

Comment:
At some point, we need to need to handle SQL-> noms conversion for UUID strings. What you have now assumes a 1:1 mapping between all the types, but that won't be good enough for UUIDs. Would add a note here.